gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -544,6 +548,31 @@ gst_tensor_trainer_epochs_is_complete (GstTensorTrainer * trainer)
   return TRUE;
 }
 
+/**
+ * @brief Wait for creating framework
+ */
+static void
+gst_tensor_trainer_wait_for_creating_framework (GstTensorTrainer * trainer)
+{
+  g_return_if_fail (trainer != NULL);
+
+  GST_INFO_OBJECT (trainer,
+      "the chain function is called before the framework is created.");
+  g_mutex_lock (&trainer->framework_created_lock);
+  if (trainer->fw_created == TRUE) {
+    /* It's already created */
+    g_mutex_unlock (&trainer->framework_created_lock);
+    return;
+  }
+
+  GST_INFO_OBJECT (trainer, "wait for framework_created_cond signal");
+  g_cond_wait (&trainer->framework_created_cond,
+      &trainer->framework_created_lock);
```

Comment:
How about adding while loop to check the if fw is created.
```suggestion
while (!trainer->fw_created)
  g_cond_wait (&trainer->framework_created_cond,
      &trainer->framework_created_lock);
```

Reply:
Thank you.!!

Reply:
Thank you.